build/stage-determine-changes.yml

```diff
@@ -0,0 +1,70 @@
+jobs:
+- job: evaluate_changes
+  displayName: 'Check for Doc Only Changes'
+  pool:
+    vmImage: 'ubuntu-latest'
+  steps:
+  - powershell: |
+      # Determine the context of the build (PR or push) and set the target branch accordingly
+      $prTargetBranch = "$(System.PullRequest.TargetBranch)"
+      $isPR = -not [string]::IsNullOrWhiteSpace($prTargetBranch)
+
+      # Normalize the target branch name for PR builds or default to 'master' for push builds
+      $targetBranchName = $isPR ? $prTargetBranch -replace 'refs/heads/', '' : "master"
+
+      Write-Host "Build context determined: $(if ($isPR) { 'Pull Request targeting ' + $targetBranchName } else { 'Push' })"
+
+      # Fetch the target or default base branch and determine the merge base
+      git fetch origin $targetBranchName
+      $mergeBase = git merge-base HEAD "origin/$targetBranchName"
+      Write-Host "Merge base with '$targetBranchName' identified at $mergeBase"
+
+      Write-Host "Comparing changes from $mergeBase..."
+      $gitDiffCommand = "git diff $mergeBase --name-only"
+      $changedFiles = Invoke-Expression $gitDiffCommand
+      $docsOnly = $false
+      $nonDocsOnly = $false
+      $mixedChanges = $false
+      $docFiles = 0
+      $nonDocFiles = 0
+
+      if ($changedFiles) {
+          Write-Host "Changed files:"
+          Write-Host $changedFiles
+      } else {
+          Write-Host "No files have changed."
+      }
+
+      foreach ($file in $changedFiles -split "`n") {
+          if ($file.EndsWith(".md") -or $file -like "*/toc.yml") {
```

Comment:
Thanks for the feedback @Xiaoy312!
I did this like that to make sure we go through all the markdown files for the linting + spell check even for the ones outside the `doc` folder for example. What do you think regarding this aspect @jeromelaban?

Reply:
Directory based is likely more interesting, but the root folder needs to be considered as well.

Reply:
@jeromelaban, @Xiaoy312 not sure I quite follow here, as the `doc` or `root` folders contain other files than markdown files. Can you give me a bit more details on what would be the needed adjustments here, please?
Waiting for these changes to be adjusted and merged to be able to replicate the same logic for Uno.Toolkit after.
Thanks a lot in advance for your feedback!

Reply:
we shouldnt really care about file types, since md/yml doesnt imply that they are strictly for docs
what matters is their location in relative to the repo
you should categorize any change under doc\ as doc change, and anywhere else is source change

in pseudo code:
```
bool srcChanged, docChanged;
foreach (var file in changedFiles)
{
	var isDoc = /* is file under `doc\` ? */;
	srcChanged |= !isDoc;
	docChanged |= isDoc;

	if (srcChanged && docChanged) break; // early exit
}
```

Reply:
@Xiaoy312, @jeromelaban I have made the adjustments for the script, let me know what do you think
`$isDoc = $file.StartsWith("doc/") -or $file -match "^.*/?[^/]*\.md$" -or $file.StartsWith(".github/")`
- `Doc` folder (self explanatory here)
- .md files at the root level (to verify the main ReadMe file, Contributing.md, Code_Of_Conduct.md, ...)
- `.github` folder as well since there are a couple of Issue and PR md templates that would be good to validate as well for markdown linting and spell checking

Reply:
looks okay for now
to be precise, `.github/**.yml` change should impact both doc and code builds

Reply:
@Xiaoy312 Here are the last adjustments
`$isDoc = $file.StartsWith("doc/") -or ($file -match "^[^/]+\.md$") -or ($file -match "^\.github/.*\.md$")`
Identifying changes as documentation if they occur:
- Within the doc folder, or
- Are Markdown files at the root level (with no subdirectories involved), or 
- Are Markdown files within the .github folder (including its subdirectories)

Reply:
LGTM